src/org/exist/client/InteractiveClient.java

```diff
@@ -2213,23 +2086,22 @@ private void connectToDatabase() {
         try {
             connect();
         } catch (final Exception cnf) {
-            if (startGUI && frame != null) {
+            if (options.startGUI && frame != null) {
                 frame.setStatus("Connection to database failed; message: " + cnf.getMessage());
             } else {
                 System.err.println("Connection to database failed; message: " + cnf.getMessage());
             }
             cnf.printStackTrace();
-            System.exit(1);
+            System.exit(SystemExitCodes.CATCH_ALL_GENERAL_ERROR_EXIT_CODE);
         }
     }
 
     /**
      * Main processing method for the InteractiveClient object
      *
-     * @param args arguments from main()
      * @return true on success, false on failure
      */
-    public boolean run(final String args[]) throws Exception {
+    public boolean run() throws Exception {
```

Comment:
```
1. `MAJOR` versions contain incompatible API changes, including changes to the on-disk format of the database;
2. `MINOR` versions add functionality or deprecate API functions, without breaking backward compatibility; and
3. `PATCH` versions contain only backwards-compatible bug fixes.
```

Because of this change next eXist version must be 4.0.0 base on rule above: `versions contain incompatible API changes`.